sky/optimizer.py

```diff
@@ -276,31 +275,40 @@ def _estimate_nodes_cost_or_time(
                             location_hint = (
                                 f' Region: {specified_resources.region}.')
 
-                    # If Kubernetes was included in the search space, then
-                    # mention "kubernetes cluster" and/instead of "catalog"
-                    # in the error message.
-                    source_hint = 'catalog'
-                    enabled_clouds = global_user_state.get_enabled_clouds()
-                    if _cloud_in_list(clouds.Kubernetes(), enabled_clouds):
-                        if specified_resources.cloud is None:
-                            source_hint = 'catalog and kubernetes cluster'
-                        elif specified_resources.cloud.is_same_cloud(
-                                clouds.Kubernetes()):
-                            source_hint = 'kubernetes cluster'
+                        # If Kubernetes was included in the search space, then
+                        # mention "kubernetes cluster" and/instead of "catalog"
+                        # in the error message.
+                        enabled_clouds = global_user_state.get_enabled_clouds()
+                        if _cloud_in_list(clouds.Kubernetes(), enabled_clouds):
+                            if specified_resources.cloud is None:
+                                source_hint = 'catalog and kubernetes cluster'
+                            elif specified_resources.cloud.is_same_cloud(
+                                    clouds.Kubernetes()):
+                                source_hint = 'kubernetes cluster'
 
                     # TODO(romilb): When `sky show-gpus` supports Kubernetes,
                     #  add a hint to run `sky show-gpus --kubernetes` to list
                     #  available accelerators on Kubernetes.
 
+                    bold = colorama.Style.BRIGHT
+                    cyan = colorama.Fore.CYAN
+                    reset = colorama.Style.RESET_ALL
+                    fuzzy_candidates_str = ''
+                    if fuzzy_candidates:
+                        fuzzy_candidates_str = (
+                            f'\nTry one of these fuzzy matches: {cyan}'
+                            f'{fuzzy_candidates}{reset}')
```

Comment:
This seems duplicated with L1035. Can we change the L1035 to be only printed when `len(task.get_resources()) > 1`?

Reply:
Thought about it. 

https://github.com/skypilot-org/skypilot/blob/58ed349c693d3dee5a5363c66dcc2dc490851147/sky/optimizer.py#L1034-L1047

has some cpu/memory hints to as logger.info. Since we don't move those hints into the exception's message, it may be fine to have the GPU candidates hint in both logger.info and exception?